sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -137,12 +139,13 @@ internal static ModelX DeserializeModelX(JsonElement element, ModelReaderWriterO
                 }
                 if (property.NameEquals("name"u8))
                 {
-                    name = property.Value.GetString();
+                    name = new(property.Value.GetString());
                     continue;
                 }
                 if (property.NameEquals("fields"u8))
                 {
-                    fields = property.Value.EnumerateArray().Select(element => element.GetString()).ToList();
+                    // TODO: May do something different depending on the semantics of expected values
```

Comment:
Collections should be clear about whether they can contain nullable values or not.  They would throw JsonException if values shouldn't be null but the JSON provides null for their values.